lib/kafka/client.rb

```diff
@@ -51,6 +52,16 @@ def get_producer(**options)
       Producer.new(cluster: @cluster, logger: @logger, **options)
     end
 
+    def get_async_producer(delivery_interval: 0, delivery_threshold: 0, **options)
```

Comment:
I never said this previously, but you have any feelings about just calling these methods `async_producer` and `producer` without the `get_` prefix? I think it's nicer.